bigtable/client/internal/endian.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bigtable/client/internal/endian.h"
+#include <assert.h>
+#include <vector>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+std::string NumericToBigEndian(std::uint64_t value) {
+  // Store big endian representation in a vector:
+  std::vector<unsigned char> bigEndian;
+  for (int index = 7; index >= 0; index--) {
+    bigEndian.push_back((value >> (8 * index)) & 0xff);
+  }
+
+  // Convert vector of chars to string:
+  std::string bigEndianString(bigEndian.begin(), bigEndian.end());
```

Comment:
Why create the vector and then the string?  Just create an string and push directly to it.  Even if you do need to create a string, why not:

```C++
return std::string(bigEndian.begin(), bigEndian.end());
```
???